android/tangram/jni/jniExports.cpp

```diff
@@ -244,6 +244,17 @@ extern "C" {
         Tangram::useCachedGlState(use);
     }
 
+    JNIEXPORT void JNICALL Java_com_mapzen_tangram_MapController_nativeCaptureSnapshot(JNIEnv* jniEnv, jobject obj, jintArray buffer) {
+        jint* arr = jniEnv->GetIntArrayElements(buffer, NULL);
+        std::vector<unsigned int> snapshot = Tangram::captureSnapshot();
+
+        for (int i = 0; i < snapshot.size(); ++i) {
+            arr[i] = snapshot[i];
+        }
+
+        jniEnv->ReleaseIntArrayElements(buffer, arr, 0);
```

Comment:
`SetIntArrayRegion()` should be able to handle this more succinctly and efficiently.